jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java

```diff
@@ -2,231 +2,284 @@
+import static com.epam.jdi.tools.PathUtils.mergePath;
+import static com.epam.jdi.tools.PrintUtils.print;
+import static com.epam.jdi.tools.PropertyReader.fillAction;
+import static com.epam.jdi.tools.PropertyReader.getProperty;
+import static com.epam.jdi.tools.ReflectionUtils.isInterface;
+import static java.lang.Boolean.parseBoolean;
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 import static org.openqa.selenium.PageLoadStrategy.*;
 
 /**
  * Created by Roman Iovlev on 14.02.2018
  * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
  */
 public class WebSettings {
-    public static ILogger logger = instance("JDI");
-    public static String getDomain() {
-        if (DRIVER.domain != null)
-            return DRIVER.domain;
+    private static WebSettings webSettings;
+    private final JDISettings jdiSettings = getJDISettings();
+
+    private WebSettings() {
+        logger = instance("JDI");
+        VISUAL_ACTION_STRATEGY = VisualCheckAction.NONE;
+        VISUAL_PAGE_STRATEGY = VisualCheckPage.NONE;
+        STRICT_SEARCH = true;
+        TEST_NAME = new Safe<>((String) null);
+        initialized = false;
+    }
+
+    public static WebSettings getWebSettings() {
+        if (webSettings == null) {
+            webSettings = new WebSettings();
+        }
+        return webSettings;
+    }
+
+    public ILogger logger;
+
+    public String getDomain() {
+        if (getJDISettings().DRIVER.domain != null)
+            return getJDISettings().DRIVER.domain;
         init();
         return "No Domain Found. Use test.properties or JDISettings.DRIVER.domain";
     }
-    public static void setDomain(String domain) {
-        DRIVER.domain = domain;
+
+    public void setDomain(String domain) {
+        getJDISettings().DRIVER.domain = domain;
     }
-    public static VisualCheckAction VISUAL_ACTION_STRATEGY = VisualCheckAction.NONE;
-    public static VisualCheckPage VISUAL_PAGE_STRATEGY = VisualCheckPage.NONE;
-    public static boolean STRICT_SEARCH = true;
-    public static boolean hasDomain() {
+
+    public VisualCheckAction VISUAL_ACTION_STRATEGY;
+    public VisualCheckPage VISUAL_PAGE_STRATEGY;
+    public boolean STRICT_SEARCH;
+
+    public boolean hasDomain() {
         init();
-        return DRIVER.domain != null && DRIVER.domain.contains("://");
+        return jdiSettings.DRIVER.domain != null && jdiSettings.DRIVER.domain.contains("://");
     }
-    public static String TEST_GROUP = "";
+
+    public String TEST_GROUP = "";
     // TODO multi properties example
-    public static Safe<String> TEST_NAME = new Safe<>((String) null);
-    public static String useDriver(JFunc<WebDriver> driver) {
-        return WebDriverFactory.useDriver(driver);
+    public Safe<String> TEST_NAME;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)